### PR TITLE
Fix inifinite loop

### PR DESCRIPTION
**Pull request by sreinhard**
This fixes an issue when where earcut enters an infinite loop. The problem is that `eliminateHole` can remove the current node that is used in `eliminateHoles`.

In he test data the outer ring is completely degenerate, so a triangulation is not really possible.

This is related to #127 and #135 (submitted by colleagues of mine), but I could not reproduce the problems reported in those issues with the current version of earcut anymore.

### CODE REVIEW

File: test/expected.json [hunk 1]
```diff
@@ -49,6 +49,7 @@
         "water-huge": 0.0011,
         "water-huge2": 0.0028,
         "bad-hole": 0.019,
+        "issue16": 4e-16,
```
Comment (sreinhard) [inline test/expected.json:52]: I had to add this to make the tests pass. I don't understand well enough why this is needed, but this seems to be reasonable?

Comment (mourner) [reply test/expected.json:52]: Can you check [this test case](https://github.com/mapbox/earcut/blob/master/test/fixtures/issue16.json) visually and see how it looks before/after? Just to make sure. The original issue for it is #16.

**Review by mourner — COMMENTED** [review]
Generally this looks great, thanks so much for working on this fix!

File: test/expected.json (hunk 1 above)
Comment (sreinhard) [reply test/expected.json:52]: Looks like just one triangle flipped for this test case.

Before | After
--- | ---
image | image

**Review by mourner — APPROVED** [review]
👍  awesome!

**Comment by mourner** [comment]
@sreinhard released this as v2.2.3. Thanks again!

**Comment by sreinhard** [comment]
Thanks @mourner for getting this in so quick!

**Pull request merged**